Source/Mock.cs

```diff
@@ -667,6 +669,44 @@ private static Expression GetValueExpression(object value, Type type)
 			return Expression.Convert(Expression.Constant(value), type);
 		}
 
+		[SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures", Justification = "By Design")]
+		internal static SetupSequencePhrase<TResult> SetupSequence<T, TResult>(Mock<T> mock, Expression<Func<T, TResult>> expression)
+			where T : class
+		{
+			if (expression.IsProperty())
+			{
+				var prop = expression.ToPropertyInfo();
+				ThrowIfPropertyNotReadable(prop);
+
+				var propGet = prop.GetGetMethod(true);
+				ThrowIfSetupExpressionInvolvesUnsupportedMember(expression, propGet);
+				ThrowIfSetupMethodNotVisibleToProxyFactory(propGet);
+
+				var setup = new SequenceMethodCall(mock, expression, propGet, new Expression[0]);
+				var targetInterceptor = GetInterceptor(((MemberExpression)expression.Body).Expression, mock);
+				targetInterceptor.AddCall(setup, SetupKind.Other);
+				return new SetupSequencePhrase<TResult>(setup);
+			}
+			else
+			{
+				var methodCall = expression.GetCallInfo(mock);
+				var targetInterceptor = GetInterceptor(methodCall.Object, mock);
+				var setup = new SequenceMethodCall(mock, expression, methodCall.Method, methodCall.Arguments.ToArray());
+				targetInterceptor.AddCall(setup, SetupKind.Other);
+				return new SetupSequencePhrase<TResult>(setup);
+			}
+		}
+
+		internal static SetupSequencePhrase SetupSequence<T>(Mock<T> mock, Expression<Action<T>> expression)
+			where T : class
```

Comment:
Same here: Does this method really have to be generic? Why not make the parameters be of types `Mock` and `LambdaExpression`? The public method in `Mock<T>` can make sure that everything is appropriately typed.